mitiq/benchmarks/volume_circuits.py

```diff
@@ -0,0 +1,112 @@
+# Copyright (C) 2022 Unitary Fund
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+"""Functions for creating circuits of the form used in quantum
+volume experiments as defined in https://arxiv.org/abs/1811.12926.
+
+Useful overview of quantum volume experiments:
+https://pennylane.ai/qml/demos/quantum_volume.html
+
+Cirq implementation of quantum volume circuits:
+cirq-core/cirq/contrib/quantum_volume/quantum_volume.py
+"""
+
+from typing import Optional, List, Tuple
+
+from numpy import random
+
+from cirq import decompose as cirq_decompose
+from cirq.circuits import Circuit
+from cirq.contrib.quantum_volume import (
+    generate_model_circuit,
+    compute_heavy_set,
+)
+from cirq.value import big_endian_int_to_bits
+
+
+from mitiq import QPROGRAM
+from mitiq.interface import convert_from_mitiq
+from mitiq.rem.measurement_result import Bitstring
+
+
+def generate_volume_circuit(
+    num_qubits: int,
+    depth: int,
+    decompose: bool = False,
+    seed: Optional[int] = None,
+    return_type: Optional[str] = None,
+) -> Tuple[QPROGRAM, List[Bitstring]]:
+    """Generate a volume circuit with the given number of qubits and depth.
+
+    The generated circuit consists of `depth` layers of random qubit
+    permutations followed by random two-qubit gates that are sampled from the
+    Haar measure on SU(4).
+
+    Args:
+        num_qubits: The number of qubits in the generated circuit.
+        depth: The number of qubits in the generated circuit.
+        decompose: Recursively decomposes the randomly sampled (numerical)
+            unitary matrix gates into simpler gates defined in cirq.ops
```

Comment:
```suggestion
            unitary matrix gates into simpler gates.
```
Better to keep it simpler. Gates can be converted differently depending on frontends.